inc/Integrations/SalesforceD2C/Auth/SalesforceD2CAuth.php

```diff
@@ -0,0 +1,200 @@
+<?php declare(strict_types = 1);
+
+namespace RemoteDataBlocks\Integrations\SalesforceD2C\Auth;
+
+use WP_Error;
+
+/**
+ * Salesforce D2C Auth class.
+ *
+ * This class is used to authenticate with Salesforce D2C using a client ID and secret.
+ */
+class SalesforceD2CAuth {
+
+	/**
+	 * Generate a token from a client ID and secret, or use an existing token if available.
+	 *
+	 * @param string $endpoint The endpoint prefix URL for the data source,
+	 * @param string $client_id The client ID (a version 4 UUID).
+	 * @param string $client_secret The client secret.
+	 * @return string|WP_Error The token or an error.
+	 */
+	public static function generate_token(
+		string $endpoint,
+		string $client_id,
+		string $client_secret
+	): string|WP_Error {
+		return self::get_saved_access_token( $client_id ) ?? self::get_token_using_client_credentials( $client_id, $client_secret, $endpoint );
+	}
+
+	/**
+	 * Get the webstores using the given endpoint, and token.
+	 *
+	 * @param string $endpoint The endpoint prefix URL for the data source.
+	 * @param string $token The token.
+	 * @return array|WP_Error The webstores or an error.
+	 */
+	public static function get_webstores(
+		string $endpoint,
+		string $token,
+	): array|WP_Error {
+		$webstores_url = sprintf( '%s/services/data/v63.0/query/?q=SELECT+name,id+from+webstore', $endpoint );
+
+		/* phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get -- We don't only work on VIP so we can't rely on that. That said, we should safely implement a wrapper that uses it when it's available. */
```

Comment:
I went back and forth on this. We could create a wrapper for this wherein if `vip_` version exists then use that or else fallback to the default one.